### ISSUE TITLE
Allow BPF enhanced session recording to work on lower kernel versions

### ISSUE THREAD

**Issue by pschisa** [gravitational/teleport#13030]
What would you like Teleport to do?
Allow for older kernel versions than 5.8+ to support BPF enhanced session recording

What problem does this solve?
Using enhanced session recording functionality on older distros

If a workaround exists, please include it.
None

Discussion:
To start, it would help to have an in-depth explanation of what functionalities are used from BPF that require the 5.8+ kernel and what advantages that offers.  Also, an answer to the following two questions would help add context:
- Why can the Teleport BPF implementation not adapt to whichever kernel is running similar to libbpf or bpftrace?
- What value does the kernel ring buffers functionality offer to Teleport?  This appears to be the functionality limiting Teleport to the 5.8 kernel in BPF.

**Comment by pschisa** [comment]
After discussion with the team, some answers to the above questions:

- Why not use bpftrace? This is how Enhanced Session Recording originally worked, we would embed programs like execsnoop, opensnoop, etc into Teleport and load them at runtime. However, this meant users had to install bcc-tools on each host which was a pain. Using BPF CO-RE we can lift this requirement.

- Ring buffer. It's a good point, we bumped the requirement because we had to increase to 5.7 for Restricted Session anyway and the ring buffer was more performant the the old perf buffer.
